cpp/src/column/column.cu

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/utilities/bit.cuh>
+#include <cudf/utilities/traits.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+
+#include <rmm/device_buffer.hpp>
+#include <rmm/mr/device_memory_resource.hpp>
+
+#include <algorithm>
+#include <vector>
+
+namespace cudf {
+
+// Copy constructor
+column::column(column const &other)
```

Comment:
What is the recommended way for a user to find out the addresses of their allocations of data, valid and children data,valid so we can do things like find out if they are managed memory or are actually an offset to a base allocation.

Reply:
Currently the only way to access pointers to a `column`'s underlying allocations is via a `column_view` or `mutable_column_view` via the `head` or `data` member functions. This is done to reduce redundancy between `column` and `column_view`. 

If that becomes too much of a burden, then I may add accessors to the `column` class itself.

Reply:
ok thats fine